services/assets.ts

```diff
@@ -0,0 +1,51 @@
+import * as Options from '../options';
+import { BaseService } from '../infrastructure';
+import { Asset, AssetData, ListAssetData } from '../models';
+
+/**
+ * A service for manipulating a Shopify shop's theme asset.
+ */
+export class Assets extends BaseService {
+    constructor(shopDomain: string, accessToken: string) {
+        super(shopDomain, accessToken, "themes");
+    }
+
+    /**
+     * Retrieves a single asset for a theme
+     * @param id Id of the theme.
+     * @param key he path to the asset within a theme.
+     * @param options Options for filtering the result.
+     */
+    public get(id: number, key: string, options?: Options.FieldOptions) {
+        return this.createRequest<AssetData>("GET", `${id}/assets.json?asset[key]=${key}&theme_id=${id}`, "", options);
```

Comment:
It's better to keep with the "style" of the rest of the package and have this line return an `Asset` (not `AssetData`), then change that empty string to the name of the JSON property which contains the object that we actually want. When creating or updating an item, that would be "asset". When listing it would be "assets". When deleting it should be an empty string.

```suggestion
        return this.createRequest<Asset>("GET", `${id}/assets.json?asset[key]=${key}&theme_id=${id}`, "asset", options);
```

Reply:
I tried it that way first, but that's how I get a error.

The api urls are `/admin/themes/#{theme_id}/assets.json` and **not** `/admin/themes/assets/#{theme_id}/assets.json`.

I have tried to set the theme id on the third parameter like this:
```
return this.createRequest<Asset>("GET", `/assets.json?asset[key]=${key}&theme_id=${id}`, `${id}`, options);
```
or this
```
return this.createRequest<Asset>("GET", `.json?asset[key]=${key}&theme_id=${id}`, `${id}/assets`, options);
```
but both had not worked